infer/src/backend/dom.ml

```diff
@@ -2173,55 +2173,78 @@ let pathset_join pname tenv (pset1: Paths.PathSet.t) (pset2: Paths.PathSet.t)
   in
   res
 
+let maximal_combine (timeout : int) (plus : 'a -> 'a -> 'a option) (xs : 'a list)  : 'a list =
```

Comment:
@rgrig: Thanks for the pull request.

Can you please add comments in the code to explain what the code is doing. For maintainability, we can only include code that people can understand and modify quickly.

Reply:
@jeremydubreil: I started adding some comments and noticed this is less complete than I thought it is. So, thanks for asking :) I should be able to update this PR before Monday.